lib/presentation/mixins/contacts_view_controller_mixin.dart

```diff
@@ -21,42 +28,83 @@ mixin class ContactsViewControllerMixin {
     const Right(ContactsInitial()),
   );
 
-  final warningBannerNotifier = ValueNotifier<WarningContactsBannerState>(
-    WarningContactsBannerState.hide,
+  final presentationPhonebookContactNotifier =
+      ValueNotifier<Either<Failure, Success>>(
+    const Right(GetPhonebookContactsInitial()),
   );
 
+  ValueNotifier<WarningContactsBannerState> get warningBannerNotifier =>
+      contactsManager.warningBannerNotifier;
+
   final FocusNode searchFocusNode = FocusNode();
 
-  final Debouncer _debouncer = Debouncer(
+  final Debouncer<String> _debouncer = Debouncer(
     const Duration(milliseconds: _debouncerIntervalInMilliseconds),
     initialValue: '',
   );
 
   final contactsManager = getIt.get<ContactsManager>();
 
-  bool get isExternalContact =>
-      textEditingController.text.isValidMatrixId &&
-      textEditingController.text.startsWith("@");
-
   void initialFetchContacts() async {
+    _refreshContacts();
     _listenContactsDataChange();
     textEditingController.addListener(() {
       _debouncer.value = textEditingController.text;
     });
 
     _debouncer.values.listen((keyword) {
-      contactsManager.searchContacts(keyword);
+      _refreshContacts();
     });
     contactsManager.initialSynchronizeContacts();
   }
 
   void _listenContactsDataChange() {
-    contactsManager.contactsStream.stream.listen(
-      (state) => presentationContactNotifier.value = state,
-    );
-    contactsManager.warningBannerStateStream.stream.listen(
-      (state) => warningBannerNotifier.value = state,
-    );
+    contactsManager.contactsNotifier.addListener(_refreshContacts);
+    contactsManager.phonebookContactsNotifier.addListener(_refreshContacts);
+  }
+
+  void _refreshContacts() {
+    final keyword = _debouncer.value;
```

Comment:
3

Reply:
I am not a fan of it. For this we always to remember to clean the `_debouncer.value`, for you, no problem because you write it. But for other people, or you in next 3 months, maybe a problem.

Reply:
but no better idea